Prevent to rename Character name on character Customization

psayafan opened the pull request:
with this you can prevent characters to rename when on Character Customization

psayafan committed:
Prevent to rename Character name on character Customization

jackpoz commented on src/server/database/Database/Implementation/CharacterDatabase.cpp:
```diff
@@ -57,6 +57,8 @@ void CharacterDatabaseConnection::DoPrepareStatements()
     PrepareStatement(CHAR_SEL_CHAR_LEVEL, "SELECT level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
     PrepareStatement(CHAR_SEL_CHAR_ZONE, "SELECT zone FROM characters WHERE guid = ?", CONNECTION_SYNCH);
     PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    PrepareStatement(CHAR_SEL_CHARACTER_DATA_NAME, "SELECT race, class, gender, `name` FROM characters WHERE guid = ?", CONNECTION_SYNCH);
```
is the field written between ` ` for some reason ?

psayafan replied:
its is my use ..... and some times its better to use ` ` in the query ... but i forget to remove this here , sorry

jackpoz commented on src/server/game/Handlers/CharacterHandler.cpp:
```diff
@@ -1405,6 +1406,13 @@ void WorldSession::HandleCharCustomize(WorldPacket& recvData)
         return;
     }
 
+    //Prevent To Rename Character (Poriya)
```
what's "Poriya" ?

psayafan replied:
it's my name :D i use it just for fast find my coding line . i should remove this too ...

jackpoz commented:
this PR has some codestyle fixes required and some engrish here and there, it's probably easier to just fix them when merging it

psayafan commented:
its my first pull request.... and i don't know what should i do now ?! i should fix this problems and pull request again ??? 
if it's possible please guide me to fix my problems thanks

Aokromes commented:
If you edit your local source and then you git push it will be added to this pr automatically.

psayafan committed:
Fix problem for PR About Prevent to rename Character name on character Customization

Carbenium commented on src/server/database/Database/Implementation/CharacterDatabase.cpp:
```diff
@@ -57,6 +57,8 @@ void CharacterDatabaseConnection::DoPrepareStatements()
     PrepareStatement(CHAR_SEL_CHAR_LEVEL, "SELECT level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
     PrepareStatement(CHAR_SEL_CHAR_ZONE, "SELECT zone FROM characters WHERE guid = ?", CONNECTION_SYNCH);
     PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    PrepareStatement(CHAR_SEL_CHARACTER_DATA_NAME, "SELECT race, class, gender, name FROM characters WHERE guid = ?", CONNECTION_SYNCH);
```
I would merge ```CHAR_SEL_CHARACTER_NAME_DATA``` and ```CHAR_SEL_CHARACTER_DATA_NAME```.
The only additional usage of ```CHAR_SEL_CHARACTER_NAME_DATA``` is in cs_character.cpp:222, so that one has to be changed.

psayafan replied:
i do so . its ok now ?!

joschiwald commented on src/server/game/Handlers/CharacterHandler.cpp:
```diff
@@ -1356,7 +1356,7 @@ void WorldSession::HandleCharCustomize(WorldPacket& recvData)
              >> customizeInfo.Face;
 
-    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_CHARACTER_NAME_DATA);
+    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_CHARACTER_DATA_NAME);
```
use characterinfo storage instead of db queries

psayafan replied:
but i think it's better . this solution is easy and simple ....

joschiwald replied:
db queries are slow

psayafan committed:
Merge CHAR_SEL_CHARACTER_NAME_DATA with CHAR_SEL_CHARACTER_DATA_NAME

L30m4nc3r commented:
for changefaction/changerace too with config. Thanks.

SnapperRy commented:
This should go under the custom config section.

psayafan committed:
Change PreventRenameCharacterOnCustomization Location in Config File

psayafan commented:
I agree the location of Config is Changed

Shauren commented on src/server/game/World/World.h:
```diff
@@ -369,6 +369,7 @@ enum WorldIntConfigs
     CONFIG_AUCTION_GETALL_DELAY,
     CONFIG_AUCTION_SEARCH_DELAY,
     CONFIG_TALENTS_INSPECTING,
+    CONFIG_PERVENT_RENAME_CUSTOMIZATION,
```
this should be under WorldBoolConfigs (and use getBoolConfig/GetBoolDefault)

Shauren commented on src/server/database/Database/Implementation/CharacterDatabase.cpp:
```diff
@@ -56,7 +56,9 @@ void CharacterDatabaseConnection::DoPrepareStatements()
     PrepareStatement(CHAR_SEL_CHAR_ZONE, "SELECT zone FROM characters WHERE guid = ?", CONNECTION_SYNCH);
-    PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level, name FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    //PrepareStatement(CHAR_SEL_CHARACTER_DATA_NAME, "SELECT race, class, gender, name FROM characters WHERE guid = ?", CONNECTION_SYNCH);
```
fully remove this prepared statement if its no longer used anywhere instead of just commenting it out

Shauren commented on src/server/game/Handlers/CharacterHandler.cpp:
```diff
@@ -1670,6 +1679,12 @@ void WorldSession::HandleCharFactionOrRaceChange(WorldPacket& recvData)
+    //Prevent To rename Character
+    if ((factionChangeInfo.Name != oldName) && (sWorld->getIntConfig(CONFIG_PERVENT_RENAME_CUSTOMIZATION) == 1))
+    {
+        SendCharFactionChange(CHAR_NAME_DECLENSION_DOESNT_MATCH_BASE_NAME, factionChangeInfo);
```
You should pick a better error code here - this one is specifically for russian clients

psayafan committed:
Change Config Type From m_int_configs To m_bool_configs for Prevent Rename Character On Customization

psayafan committed:
Change CHAR_NAME_DECLENSION_DOESNT_MATCH_BASE_NAME Error To CHAR_NAME_FAILURE

psayafan commented:
Thanks @Shauren i do all things that you said <3

myran2 commented on src/server/game/World/World.h:
```diff
@@ -175,6 +175,7 @@ enum WorldBoolConfigs
     CONFIG_HOTSWAP_BUILD_FILE_RECREATION_ENABLED,
     CONFIG_HOTSWAP_INSTALL_ENABLED,
     CONFIG_HOTSWAP_PREFIX_CORRECTION_ENABLED,
+    CONFIG_PERVENT_RENAME_CUSTOMIZATION,
```
PERVENT -> PREVENT?

DDuarte commented on src/server/database/Database/Implementation/CharacterDatabase.h:
```diff
@@ -56,6 +56,7 @@ enum CharacterDatabaseStatements
     CHAR_SEL_CHAR_LEVEL,
     CHAR_SEL_CHAR_ZONE,
     CHAR_SEL_CHARACTER_NAME_DATA,
+    CHAR_SEL_CHARACTER_DATA_NAME,
```
Remove line

DDuarte commented on src/server/database/Database/Implementation/CharacterDatabase.cpp:
```diff
@@ -56,7 +56,8 @@ void CharacterDatabaseConnection::DoPrepareStatements()
     PrepareStatement(CHAR_SEL_CHAR_ZONE, "SELECT zone FROM characters WHERE guid = ?", CONNECTION_SYNCH);
-    PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level, name FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    
```
Remove line

DDuarte commented on src/server/worldserver/worldserver.conf.dist:
```diff
@@ -3739,4 +3749,4 @@ Metric.ConnectionInfo = "127.0.0.1;8086;worldserver"
 
 #
-###################################################################################################
+###################################################################################################
```
Do not remove the empty line at the end of the file.

DDuarte commented on src/server/worldserver/worldserver.conf.dist:
```diff
@@ -3029,6 +3029,16 @@ NoGrayAggro.Above = 0
+#                     1 - Enable(Can't Rename on Character Customizations)
+#
+
+PreventRenameCharacterOnCustomization = 0
```
The language needs to be improved. @tkrokli could you give a hand?

psayafan replied:
yes if possible help me about this .

tkrokli replied:
Please check if I have kept the original and intended meaning of the text. I have tried to follow the standards used in worldserver.conf.dist (simple and straight English text pertaining to the source):
```
#
#    PreventRenameCharacterOnCustomization
#        Description: If option is set to 1, player can not rename the character in character customization.
#                     Applies to all character customization commands.
#        Default:     0 - (Disabled, character can be renamed in Character Customization)
#                     1 - (Enabled, character can not be renamed in Character Customization)
#

PreventRenameCharacterOnCustomization = 0

#
```

prestonjamesparsons replied:
^this

psayafan committed:
Fix Some Stupid Problems :D thanks to DDuarte

GitVerge commented on src/server/game/Handlers/CharacterHandler.cpp:
```diff
@@ -1405,6 +1406,13 @@ void WorldSession::HandleCharCustomize(WorldPacket& recvData)
     }
 
+    //Prevent To Rename Character
+    if ((customizeInfo.Name != plrName) && (sWorld->getBoolConfig(CONFIG_PERVENT_RENAME_CUSTOMIZATION)))
```
travis, "PERVENT"

psayafan committed:
Its Done Done :| Prevent !

Shauren commented on src/server/database/Database/Implementation/CharacterDatabase.cpp:
```diff
@@ -56,7 +56,7 @@ void CharacterDatabaseConnection::DoPrepareStatements()
     PrepareStatement(CHAR_SEL_CHAR_LEVEL, "SELECT level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
     PrepareStatement(CHAR_SEL_CHAR_ZONE, "SELECT zone FROM characters WHERE guid = ?", CONNECTION_SYNCH);
-    PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level FROM characters WHERE guid = ?", CONNECTION_SYNCH);
+    PrepareStatement(CHAR_SEL_CHARACTER_NAME_DATA, "SELECT race, class, gender, level, name FROM characters WHERE guid = ?", CONNECTION_SYNCH);
```
When I said remove - it also meant removing the value from CharacterDatabase.h

psayafan replied:
i'm sorry i forgot

psayafan committed:
re write the config file text thanks to tkrokli

prestonjamesparsons commented on src/server/game/Handlers/CharacterHandler.cpp:
```diff
@@ -1670,6 +1679,12 @@ void WorldSession::HandleCharFactionOrRaceChange(WorldPacket& recvData)
         return;
     }
 
+    //Prevent To rename Character (Poriya)
```
And this ;)

psayafan replied:
i Removed this before ! its OK now ...

psayafan commented:
it's OK now ?! is there any problem in this ?! 
and what is should do now ? :D

psayafan committed:
Fix and match with Config Default Values

0 = Character Can Rename on Character Customization
1 = Character Can't Rename on Character Customization

psayafan committed:
no not Nevermind the oldest is true :|

tkrokli commented:
Well, you could follow up Shauren's comment about CharacterDatabase.h

psayafan commented on src/server/game/World/World.h:
```diff
@@ -175,7 +175,7 @@ enum WorldBoolConfigs
     CONFIG_HOTSWAP_INSTALL_ENABLED,
     CONFIG_HOTSWAP_PREFIX_CORRECTION_ENABLED,
-    CONFIG_PERVENT_RENAME_CUSTOMIZATION,
+    CONFIG_PREVENT_RENAME_CUSTOMIZATION,
```
i did before @tkrokli

psayafan commented on src/server/database/Database/Implementation/CharacterDatabase.h:
```diff
@@ -56,7 +56,6 @@ enum CharacterDatabaseStatements
     CHAR_SEL_CHAR_LEVEL,
     CHAR_SEL_CHAR_ZONE,
     CHAR_SEL_CHARACTER_NAME_DATA,
-    CHAR_SEL_CHARACTER_DATA_NAME,
```
i did before @tkrokli

tkrokli replied:
Oh. I see. I just did not see it before and it did not look like the file name was listed in your PR.

psayafan replied:
it's OK thanks for your attention and comment
